### PR TITLE
fix: Line of route - BODS data (Merseytravel)

### PR DESCRIPTION
**Pull request by alainajaffer**

### CODE REVIEW

File: site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx [hunk 1]
```diff
@@ -89,6 +90,56 @@ const getMode = (vehicleMode: Modes | VehicleMode) => {
     return mode;
 };
 
+const groupByJourneyPattern = (routesData: RouteWithServiceInfoPreformatted[]): RouteWithServiceInfo[] => {
```
Comment (DeannaSharma) [inline site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx:93]: There's quite a bit going on in this function, I would suggest brining this out into the utils file and making a unit test to check it correctly formats it as we expect. And if we know of any edge cases to test how this function would handle those edge cases.

File: site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx [hunk 2]
```diff
@@ -145,7 +196,7 @@ const CreateConsequenceServices = (props: CreateConsequenceServicesProps): React
 
     useEffect(() => {
         const loadOptions = async () => {
-            const serviceDataToShow: RouteWithServiceInfo[] = [];
+            const serviceDataToShow: RouteWithServiceInfo[] | RouteWithServiceInfoPreformatted[] = [];
```
Comment (DeannaSharma) [inline site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx:199]:
```suggestion
            const serviceDataToShow: RouteWithServiceInfoPreformatted[] = [];
```

Can we update the type like so, then on line 226 and 236 we don't have to cast the type. I've tested it locally and looks like the map still works, not sure if there was a specific reason for including both types?

File: site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx [hunk 3]
```diff
@@ -89,6 +90,56 @@ const getMode = (vehicleMode: Modes | VehicleMode) => {
...
+    routesData.forEach((data) => {
+        const groupedData: RouteWithServiceInfo = {
+            outbound: {},
+            inbound: {},
+            serviceId: data.serviceId,
+            serviceCode: data.serviceCode,
+            lineId: data.lineId,
+        };
+
+        if (Array.isArray(data.outbound)) {
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx]: done

File: site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx (hunk 1 above)
Comment (alainajaffer) [reply site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx]: done

File: site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx (hunk 2 above)
Comment (alainajaffer) [reply site/pages/create-consequence-services/[disruptionId]/[consequenceIndex].page.tsx]: done

**Review by DeannaSharma — APPROVED** [review]
